<?xml version="1.0"?>
<launch>
<<<<<<< HEAD
    <let name="mono3d_flag" value="true"/>
    <let name="mono3d_ckpt_file" value="/home/yliuhb/vision_collection/model/det3d/mono3d_yolox_576_768.onnx"/>
    <let name="mono3d_gpu_index" value="0"/>

    <let name="seg_flag" value="true"/>
    <let name="seg_ckpt_file" value="/home/yliuhb/vision_collection/model/segmentation/bisenetv1.onnx"/>
    <let name="seg_gpu_index" value="0"/>
    
    <let name="monodepth_flag" value="true"/>
    <let name="monodepth_ckpt_file" value="/home/yliuhb/vision_collection/model/monodepth/monodepth_res101_384_1280.onnx"/>
    <let name="monodepth_gpu_index" value="0"/>
=======
    <arg name="mono3d_flag" default="true"/>
    <arg name="mono3d_ckpt_file" value="/home/yliuhb/vision_collection/model/det3d/mono3d_yolox_576_768.onnx"/>

    <arg name="seg_flag" default="true"/>
    <arg name="seg_ckpt_file" default="/home/yliuhb/vision_collection/model/segmentation/bisenetv1.onnx"/>
    
    <arg name="monodepth_flag" default="true"/>
    <arg name="monodepth_ckpt_file" default="/home/yliuhb/vision_collection/model/monodepth/monodepth_res101_384_1280.onnx"/>
>>>>>>> 3166b3a (fix bugs in command line configuration; add QOS profile example; add support for compressed images)
    
    <arg name="camera_topic" default="/nuscenes/CAM_FRONT/image"/>
    <arg name="compressed_image_topic" default="/sensing/camera/camera0/image_rect_color/compressed"/>
    <arg name="camera_param" default="/nuscenes/CAM_FRONT/camera_info"/>
    <arg name="output_bbox_topic" default="/bboxes"/>
    <arg name="output_seg_pointcloud_topic" default="/point_cloud"/>

    <node pkg="ros2_vision_inference" exec="vision_inference_node" name="vision_inference" output="screen">
        <param name="MONO3D_FLAG" value="$(var mono3d_flag)"/>
        <param name="MONO3D_CKPT_FILE" value="$(var mono3d_ckpt_file)"/>
        <param name="MONO3D_GPU_INDEX" value="$(var mono3d_gpu_index)"/>

        <param name="SEG_FLAG" value="$(var seg_flag)"/>
        <param name="SEG_CKPT_FILE" value="$(var seg_ckpt_file)"/>
        <param name="SEG_GPU_INDEX" value="$(var seg_gpu_index)"/>

        <param name="MONODEPTH_FLAG" value="$(var monodepth_flag)"/>
        <param name="MONODEPTH_CKPT_FILE" value="$(var monodepth_ckpt_file)"/>
        <param name="MONODEPTH_GPU_INDEX" value="$(var monodepth_gpu_index)"/>

        <remap from="/image_raw" to="$(var camera_topic)"/>
        <remap from="/compressed_image" to="$(var compressed_image_topic)"/>
        <remap from="/camera_info" to="$(var camera_param)"/>
        <remap from="/bboxes" to="$(var output_bbox_topic)"/>
        <remap from="/point_cloud" to="$(var output_seg_pointcloud_topic)"/>
    </node>

</launch>